Many improvements to evolutionSteer !

alfred1877 opened the pull request:
Hey Carykh !

First of all i wan't to thank you for all your work. I am one of your silent admirers on Youtube and always watch your videos with great interest!

I downloaded the **evolutionSteer** project in order to create my first creatures and went addicted with it (!). I was frustrated to see only jumping creatures so worked on it to improve the selection mechanism, force them to walk and eventually conquer the world. This is the result of my work, the best i did was a creature who did 29 chomps in 2 minutes (with 90° curving). You can see it softly walking [here](https://www.youtube.com/watch?v=RZWCME8O2Rg) with its 6 feet.

Here is a (non-exhaustive) list of modifications i made to your original script:
* added **Radioactive mode** to temporarily increase the mutation rate of the creatures
* added **gift function**, that increases the duration of the simulation based on the number of chomps
* selection features to **remove jumpers** !!! (i hate them) and to increase the brain width during evolution
* possibility to change the number of creatures per generation (a smaller number such as 250 creatures seems to have a better efficiency for evolution)
* **multithreading** support, based on the work by [nathan29299292](https://github.com/nathan29299292/) and [d4rkc0d3r](https://github.com/d4rkc0d3r/)
* **saving/autosaving** function, initially based on [darkracer](https://github.com/darkracer)'s work (but i did eventually rewrote everything for heap memory limitations). It would allow to exchange creatures between people which is great !
* various **speed improvements**, especially one based on the [ticket #10 by MauroMombelli](https://github.com/carykh/evolutionSteer/issues/10)
* *Processing 3* compatibility
* Well and sort-of a documentation ;-)

Well, i hope you will find these modifications interesting. While i wrote this message a new species has emerged, i think i owe them to go and feed them with little multicolour blobs...

Have a nice day!

Joseph

Timfa2112 committed:
Add extra neuron for heartbeat, add heartbeat

Note: This also causes an extra unneccesary neuron to be created in the output layer, but this should be harmless.

alfred1877 committed:
Create README.md

alfred1877 committed:
Update README.md

jbenzakoun committed:
Added radioactive mode

Boosts mutation by pressing "r"

jbenzakoun committed:
Merge remote-tracking branch 'origin/master'

jbenzakoun committed:
Fresh blood

Create new creatures in radioactive mode

jbenzakoun committed:
Simulation timer

Easier setting for simulation timer. Bonus time when chomping.

jbenzakoun committed:
Creature number

Easy change for creature number

jbenzakoun committed:
Processing 3

Compatibility

jbenzakoun committed:
Moved global variables

From nathan29299292

jbenzakoun committed:
Added timer

From nathan29299292

jbenzakoun committed:
Multithread

And modification of many classes for thread compatibility

jbenzakoun committed:
Other technique to remove jumpers

I hate jumpers

jbenzakoun committed:
Mass extinction

Just press 'k'

jbenzakoun committed:
Adding gitattributes for EOL

jbenzakoun committed:
Multithread 2

SImplified multithread

jbenzakoun committed:
Savable

issued from darkracer branch

jbenzakoun committed:
Save screen

Added save and load screens

jbenzakoun committed:
Interface glitch

COmpatibility between radioactivity and save

jbenzakoun committed:
Autosave

Added autosave capabilities

jbenzakoun committed:
Autosave 2

Bug fix

jbenzakoun committed:
Autosave timecoded

To create various files with timecode

jbenzakoun committed:
AutoPause

Little tweak allows you to leave your computer

jbenzakoun committed:
Saving 2 

Update saving capabilities (rule out memory heap size problem)

jbenzakoun committed:
Optimisation

Approximate sigmoid function for speed improvement

jbenzakoun committed:
Expanded brain

Can add one layer when mutating

jbenzakoun committed:
Ignore all gz files

jbenzakoun committed:
Light save

jbenzakoun committed:
Gift function

jbenzakoun committed:
Update mutation comportement

loss function for large brains

jbenzakoun committed:
Changed default parameters

jbenzakoun committed:
Update readme file

jbenzakoun committed:
Readme update for install

jbenzakoun committed:
added commands for reducing and increasing gift

jbenzakoun committed:
added "k" in readme

jbenzakoun committed:
Modified mutation properties

Won't mutate brain and body at the same time

MauroMombelli commented:
BEAUTIFUL!!

i like how you cut the sigmoid in diferent linear approximation. another
inprovement could be to do math in integer, or use gpu.

about thread, would be nice a way to set core-1 thread, so we can keep one
free for normal stuff.

2017-07-27 19:27 GMT+02:00 alamaz <notifications@github.com>:

> Hey Carykh !
>
> First of all i wan't to thank you for all your work. I am one of your
> silent admirers on Youtube and always watch your videos with great interest!
>
> I downloaded the *evolutionSteer* project in order to create my first
> creatures and went addicted with it (!). I was frustrated to see only
> jumping creatures so worked on it to improve the selection mechanism, force
> them to walk and eventually conquer the world. This is the result of my
> work, the best i did was a creature who did 29 chomps in 2 minutes (with
> 90° curving). You can see it softly walking here
> <https://www.youtube.com/watch?v=RZWCME8O2Rg> with its 6 feet.
>
> Here is a (non-exhaustive) list of modifications i made to your original
> script:
>
>    - added *Radioactive mode* to temporarily increase the mutation rate
>    of the creatures
>    - added *gift function*, that increases the duration of the simulation
>    based on the number of chomps
>    - selection features to *remove jumpers* !!! (i hate them) and to
>    increase the brain width during evolution
>    - possibility to change the number of creatures per generation (a
>    smaller number such as 250 creatures seems to have a better efficiency for
>    evolution)
>    - *multithreading* support, based on the work by nathan29299292
>    <https://github.com/nathan29299292/> and d4rkc0d3r
>    <https://github.com/d4rkc0d3r/>
>    - *saving/autosaving* function, initially based on darkracer
>    <https://github.com/darkracer>'s work (but i did eventually rewrote
>    everything for heap memory limitations). It would allow to exchange
>    creatures between people which is great !
>    - various *speed improvements*, especially one based on the ticket #10
>    by MauroMombelli <https://github.com/carykh/evolutionSteer/issues/10>
>    - *Processing 3* compatibility
>    - Well and sort-of a documentation ;-)
>
> Well, i hope you will find these modifications interesting. While i wrote
> this message a new species has emerged, i think i owe them to go and feed
> them with little multicolour blobs...
>
> Have a nice day!
>
> Joseph
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/carykh/evolutionSteer/pull/13
> Commit Summary
>
>    - Create README.md
>    - Update README.md
>    - Added radioactive mode
>    - Merge remote-tracking branch 'origin/master'
>    - Fresh blood
>    - Simulation timer
>    - Creature number
>    - Processing 3
>    - Moved global variables
>    - Added timer
>    - Multithread
>    - Other technique to remove jumpers
>    - Mass extinction
>    - Adding gitattributes for EOL
>    - Multithread 2
>    - Savable
>    - Save screen
>    - Interface glitch
>    - Autosave
>    - Autosave 2
>    - Autosave timecoded
>    - AutoPause
>    - Saving 2
>    - Optimisation
>    - Expanded brain
>    - Ignore all gz files
>    - Light save
>    - Gift function
>    - Update mutation comportement
>    - Changed default parameters
>    - Update readme file
>    - Readme update for install
>    - added commands for reducing and increasing gift
>    - added "k" in readme
>    - Modified mutation properties
>
> File Changes
>
>    - *A* .gitattributes
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-0> (1)
>    - *A* .gitignore
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-1> (7)
>    - *M* Axon.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-2> (29)
>    - *M* Brain.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-3> (140)
>    - *M* Creature.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-4> (421)
>    - *M* Muscle.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-5> (73)
>    - *M* Node.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-6> (153)
>    - *A* README.md
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-7> (56)
>    - *A* code/jackson-core-2.9.0.pr4.jar
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-8> (0)
>    - *A* code/jackson-dataformat-smile-2.9.0.pr4.jar
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-9> (0)
>    - *A* computingThread.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-10> (31)
>    - *M* evolutionSteer.pde
>    <https://github.com/carykh/evolutionSteer/pull/13/files#diff-11> (904)
>
> Patch Links:
>
>    - https://github.com/carykh/evolutionSteer/pull/13.patch
>    - https://github.com/carykh/evolutionSteer/pull/13.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/carykh/evolutionSteer/pull/13>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGE80dH6x-IrVD6Ng7yFJQVqooXUccOwks5sSMiBgaJpZM4OlmOU>
> .
>

jbenzakoun committed:
Mass extinction percentage

Can choose the percentage for KILL func

jbenzakoun committed:
Brain improvement

Full height and rewritten sigmoid curve

jbenzakoun committed:
Chomp limitation 

to 100

jbenzakoun committed:
Save v4

Simplified save files

jbenzakoun committed:
Brainiac mode

Activate by pressing 'b'

jbenzakoun committed:
Variable friction

According to neural network

jbenzakoun committed:
Save v4b

bugfix in v4 loading

jbenzakoun committed:
Updated readme for Brainiac

jbenzakoun committed:
Bugfix

Removed potential crash sources

jbenzakoun committed:
maxChomp

Variable max number of chomps

jbenzakoun committed:
Jumper truncature

Remove jumper limitation when the gift is < than 12 sec

jbenzakoun committed:
PowerDouble

added powerdouble function (cf readme)

jbenzakoun committed:
Heartbeat neuron

Courtesy of Master2112

jbenzakoun committed:
Merge remote-tracking branch 'Master2112/master'

jbenzakoun committed:
Linear transform for brain 

Allows to create imilar brains with more neurons

jbenzakoun committed:
Complex fitness improvement

Fitness is defined by speed and chomps
Food angulation is increased progressively so that evolution naturally leads to steering
Supplemental heartbeat, faster
Jumper truncation removed

hanss314 commented:
If you haven't noticed by now, cary is bad at maintaining his github, you might want to contact him via discord or twitter DM

beld0n commented:
hey guys!
im new to github, but i really like this game
so my question is, how can i add all of this changes to my file easily? without changing all the code myself, because im too dumb for that
thank you :)

nzifnab commented on Brain.pde:
```diff
@@ -80,50 +83,106 @@ class Brain {
     }
   }
+  public float linear(float input){
+    if(input >= -5 || input <= 5){
```
On this line in windows x64 I get the error: "processing.app.SketchException: Not expecting symbol 0xA0, which is NO-BREAK SPACE."

The offending character appears to be the whitespace between the or (`||`) and the second use of the `input` variable.

Deleting it and hitting the spacebar appears to resolve this issue.

calvinjuarez commented:
@alamaz Your video looks to be unavailable and I wanna see!
